### DISCUSSION TITLE
important selector and rtl modifier

### DISCUSSION THREAD

**Discussion by y1n0** [tailwindlabs/tailwindcss#6756] [Help]
Hello, I noticed that when using a selector as an important strategy and the rtl modifer, classes set to the element that has the `dir` attributes do not take effect, that is because the generated style rules look like something like this: 
```css
#app [dir=rtl] .rtl\:border-r {
  // ...
}
```
which requires 3 levels of nesting for the class to take effect.
I have used [tailwindcss-dir](https://github.com/RonMelkhior/tailwindcss-dir) plugin before, and it did not have this issue because it generated rules that combine the `dir` selector with the class: 
```css
#app [dir="rtl"] .rtl:border-r,
#app [dir="rtl"].rtl\:border-r {
  // ...
}
```
and that worked fine.

Here is a reproduction what I am talking about https://play.tailwindcss.com/R5VqKGHY1Z. notice how `rtl:border-r-8` does not apply to h1 because it is not a child of an element with `dir="rtl"` which itself should be a child an element `id="app"`.

**Comment by Thomas9878** [comment]
I'm other than defying the issue, in the event that no one genuinely minds in any case, help me with settling this issue.
[OneVanilla.com](https://www.onevanilla.top/)
